qa/integration/specs/monitoring_api_spec.rb

```diff
@@ -245,6 +245,33 @@
     logging_get_assert logstash_service, "INFO", "TRACE"
   end
 
+  it "should retrieve the pipeline flow statuses" do
+    logstash_service = @fixture.get_service("logstash")
+    logstash_service.start_with_stdin
+    logstash_service.wait_for_logstash
+    number_of_events.times { logstash_service.write_to_stdin("Testing flow metrics") }
+
+    Stud.try(max_retry.times, [StandardError, RSpec::Expectations::ExpectationNotMetError]) do
+      # node_stats can fail if the stats subsystem isn't ready
+      result = logstash_service.monitoring_api.node_stats rescue nil
+      expect(result).not_to be_nil
+      # we use fetch here since we want failed fetches to raise an exception
+      # and trigger the retry block
+      flow_status = result.fetch("pipelines").fetch("main").fetch("flow")
+      expect(flow_status).not_to be_nil
+      expect(flow_status["worker_concurrency"]["lifetime"]).not_to be_nil
+      expect(flow_status["worker_concurrency"]["current"]).not_to be_nil
+      expect(flow_status["input_throughput"]["lifetime"]).to eq(number_of_events)
```

Comment:
These flow metrics are a rate relative to pipeline uptime, so unless the execution of the inputs took _exactly_ 1s, we aren't going to get exactly `number_of_events` out (and without mocking the clock, there will likely be a lot of jitter from run to run. We can use the [`be_within` matcher](https://relishapp.com/rspec/rspec-expectations/docs/built-in-matchers/be-within-matcher) if we have a realistic target, or can make a [compound expectation](https://relishapp.com/rspec/rspec-expectations/v/3-11/docs/compound-expectations) if we have a specific ranges.

Reply:
the good news is now flow metrics are captured after giving some window to poller to capture. And yes, since it is the rate `input_throughput` never gets the `number_of_events`. It really depends on how `@fixture.get_service("logstash")` behaves but I think we can set a condition against zero, meaning integration tests transmitted events so that metrics (except backpressure here) never becomes zero. How do you think?